lib/tpm2_options.c

```diff
@@ -424,13 +429,25 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv,
 out:
 
     if (show_help) {
-        bool did_manpager = execute_man(argv[0]);
-        if (!did_manpager) {
+        bool did_manpager = true;
+        bool is_manpage = true;
+        bool is_short = true;
+        if (help_arg) {
+            is_manpage = strcmp(help_arg,"manpage");
+            is_short = strcmp(help_arg,"summary");
+        }
+        if (!is_manpage || help_arg == NULL) {
```

Comment:
if it's not manpage, execute the manpager? Seems wrong....

Reply:
yeah, it seems wrong, but the idea was to leave the manpage as default, if the -h or --help are invoked with no arguments, then the manpage has the priority but now that you mention maybe the argument manpage is not necessary